File: MS/src/main/java/com/yskc/ms/entity/rs/TechRequirement.java
package com.yskc.ms.entity.rs;

import com.baomidou.mybatisplus.annotations.TableName;

import java.math.BigDecimal;
import java.util.Date;

/**
 * @BelongsProject: ms
 * @BelongsPackage: com.yskc.ms.entity.ms
 * @Author: zhangdingfu
 * @CreateTime: 2020-05-21 09:17
 * @Description: 技术需求
 */
@TableName("tech_requirement")
public class TechRequirement {

    private Integer id;
    private Integer companyId;
    private String linkName;
    private String position;
    private String linkTel;
    private String linkEmail;
    private String description;
    private String filePath;
    private String techName;
    private String domain;
    private BigDecimal investment;
    private String requirement;
    private String cooperateType;
    private String otherDomain;
    private String otherCooperateType;
    private Integer year;
    private Integer creatorId;
    private Integer lastUpdatorId;
    private Date createTime;
    private Date lastUpdateTime;
    private Integer msCreatorId;
    private Integer msLastUpdatorId;
    private Integer status;
    private String cooperateSchool;//合作院校

    public String getCooperateSchool() {
        return cooperateSchool;
    }

    public void setCooperateSchool(String cooperateSchool) {
        this.cooperateSchool = cooperateSchool;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Integer getCompanyId() {
        return companyId;
    }

    public void setCompanyId(Integer companyId) {
        this.companyId = companyId;
    }

    public String getLinkName() {
        return linkName;
    }

    public void setLinkName(String linkName) {
        this.linkName = linkName;
    }

    public String getPosition() {
        return position;
    }

    public void setPosition(String position) {
        this.position = position;
    }

    public String getLinkTel() {
        return linkTel;
    }

    public void setLinkTel(String linkTel) {
        this.linkTel = linkTel;
    }

    public String getLinkEmail() {
        return linkEmail;
    }

    public void setLinkEmail(String linkEmail) {
        this.linkEmail = linkEmail;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getFilePath() {
        return filePath;
    }

    public void setFilePath(String filePath) {
        this.filePath = filePath;
    }

    public String getTechName() {
        return techName;
    }

    public void setTechName(String techName) {
        this.techName = techName;
    }

    public String getDomain() {
        return domain;
    }

    public void setDomain(String domain) {
        this.domain = domain;
    }

    public BigDecimal getInvestment() {
        return investment;
    }

    public void setInvestment(BigDecimal investment) {
        this.investment = investment;
    }

    public String getRequirement() {
        return requirement;
    }

    public void setRequirement(String requirement) {
        this.requirement = requirement;
    }

    public String getCooperateType() {
        return cooperateType;
    }

    public void setCooperateType(String cooperateType) {
        this.cooperateType = cooperateType;
    }

    public Integer getCreatorId() {
        return creatorId;
    }

    public void setCreatorId(Integer creatorId) {
        this.creatorId = creatorId;
    }

    public Integer getLastUpdatorId() {
        return lastUpdatorId;
    }

    public void setLastUpdatorId(Integer lastUpdatorId) {
        this.lastUpdatorId = lastUpdatorId;
    }

    public Date getCreateTime() {
        return createTime;
    }

    public void setCreateTime(Date createTime) {
        this.createTime = createTime;
    }

    public Date getLastUpdateTime() {
        return lastUpdateTime;
    }

    public void setLastUpdateTime(Date lastUpdateTime) {
        this.lastUpdateTime = lastUpdateTime;
    }

    public Integer getYear() {
        return year;
    }

    public void setYear(Integer year) {
        this.year = year;
    }

    public String getOtherDomain() {
        return otherDomain;
    }

    public void setOtherDomain(String otherDomain) {
        this.otherDomain = otherDomain;
    }

    public String getOtherCooperateType() {
        return otherCooperateType;
    }

    public void setOtherCooperateType(String otherCooperateType) {
        this.otherCooperateType = otherCooperateType;
    }

    public Integer getMsCreatorId() {
        return msCreatorId;
    }

    public void setMsCreatorId(Integer msCreatorId) {
        this.msCreatorId = msCreatorId;
    }

    public Integer getMsLastUpdatorId() {
        return msLastUpdatorId;
    }

    public void setMsLastUpdatorId(Integer msLastUpdatorId) {
        this.msLastUpdatorId = msLastUpdatorId;
    }

    public Integer getStatus() {
        return status;
    }

    public void setStatus(Integer status) {
        this.status = status;
    }
}

File: YSDB/RSDB/Models/tech_requirement.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace RSDB.Models
{
    /// <summary>
    ///技术需求表
    /// </summary>
    public class tech_requirement : tablebase
    {
        [Key]
        public int id { get; set; }

        public int companyId { get; set; }
        /// <summary>
        /// 联系人
        /// </summary>
        [Required, MaxLength(20)]
        public string linkName { get; set; }
        /// <summary>
        /// 职务
        /// </summary>
        [Required, MaxLength(50)]
        public string position { get; set; }
        /// <summary>
        ///  联系电话
        /// </summary>
        [Required, MaxLength(50)]
        public string linkTel { get; set; }
        /// <summary>
        ///  联系邮箱
        /// </summary>
        [Required, MaxLength(50)]
        public string linkEmail { get; set; }
        /// <summary>
        ///  简介
        /// </summary>
        [Required, MaxLength(300)]
        public string description { get; set; }
        /// <summary>
        ///  附件地址
        /// </summary>
        [Required, MaxLength(500)]
        public string filePath { get; set; }
        /// <summary>
        ///  技术需求名称
        /// </summary>
        [Required, MaxLength(300)]
        public string techName { get; set; }

        /// <summary>
        ///  技术领域
        /// </summary>
        [Required, MaxLength(50)]
        public string domain { get; set; }

        /// <summary>
        ///  投资金额
        /// </summary> 
        [Column(TypeName = "decimal(18,2)")]
        public decimal investment { get; set; }
        /// <summary>
        ///  需求描述
        /// </summary>
        [Required, MaxLength(300)]
        public string requirement { get; set; }

        /// <summary>
        ///  合作方式
        /// </summary>
        [Required, MaxLength(50)]
        public string cooperateType { get; set; }

        public int year { get; set; }

        /// <summary>
        ///  其他领域
        /// </summary>
        [MaxLength(100)]
        public string otherDomain { get; set; }
        /// <summary>
        ///  其他合作方式
        /// </summary>
        [MaxLength(100)]
        public string otherCooperateType { get; set; }
        /// <summary>
        ///  状态:0 普通, 1 作废
        /// </summary>
        public int status { get; set; }
        /// <summary>
        ///  合作院校
        /// </summary>
        [MaxLength(50)]
        public string cooperateSchool { get; set; }
    }
}
